src/AzureAuth/CommandMain.cs

```diff
@@ -255,11 +255,21 @@ private int GetToken()
         {
             try
             {
-                ITokenFetcher tokenFetcher = this.TokenFetcher();
-                TokenResult tokenResult = (this.tokenFetcherOptions.Scopes == null
+                TokenResult tokenResult = null;
+                ITokenFetcher tokenFetcher = null;
+
+                // When running multiple AzureAuth processes with the same tenant and resource ID,
+                // They may prompt many times, which is annoying and unexpected.
+                // Use GlobalLock to ensure that only one process can access the corresponding resource at the same time.
+                string lockName = "Global\\" + this.Tenant + this.Resource;

```

Comment:
I thought the lock needed to be around the `Resource`, `Client`, and `Tenant` trio. Is that right, @kyle-rader?

Reply:
Yep, great catch - we only want to synchronize calls that will benefit from one another's caching. Which is the whole trio.